pkg/storegateway/bucket.go

```diff
@@ -1481,29 +1462,24 @@ func storeCachedLabelValues(ctx context.Context, indexCache indexcache.IndexCach
 	indexCache.StoreLabelValues(ctx, userID, blockID, labelName, entry.MatchersKey, data)
 }
 
-// them up by downsampling resolution and allows querying.
 // bucketBlockSet holds all blocks of an equal label set. It internally splits
+// them up by downsampling resolution and allows querying.
 type bucketBlockSet struct {
-	labels      labels.Labels
 	mtx         sync.RWMutex
 	resolutions []int64          // Available resolution, high to low (in milliseconds).
 	blocks      [][]*bucketBlock // Ordered buckets for the existing resolutions.
 }
 
 // newBucketBlockSet initializes a new set with the known downsampling windows hard-configured.
 // The set currently does not support arbitrary ranges.
-func newBucketBlockSet(lset labels.Labels) *bucketBlockSet {
+func newBucketBlockSet() *bucketBlockSet {
 	return &bucketBlockSet{
-		labels:      lset,
 		resolutions: []int64{downsample.ResLevel2, downsample.ResLevel1, downsample.ResLevel0},
 		blocks:      make([][]*bucketBlock, 3),
 	}
 }
 
 func (s *bucketBlockSet) add(b *bucketBlock) error {
-	if !labels.Equal(s.labels, labels.FromMap(b.meta.Thanos.Labels)) {
-		return errors.New("block's label set does not match set")
-	}
```

Comment:
sorry if i'm missing something obvious, but why is this check not necessary anymore?

Reply:
`s.labels` is originally set from `labels.FromMap(b.meta.Thanos.Labels)`, but the latter appears to always be empty, as according to https://github.com/grafana/mimir/issues/2021 "we don't inject external labels in the query response". When running the development docker compose it seems to be always empty as well.

Reply:
In this PR we're removing the support for multiple block sets, which means all blocks are loaded into the same set. Because of this, we shouldn't run this check anymore, because it's expected that all blocks are added to the same `bucketBlockSet`.

From this perspective, this change LGTM.